contrib/aerospike/aerospike-client.v5/aerospike.go

```diff
@@ -0,0 +1,420 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+// Package aerospike provides functions to trace the aerospike/aerospike-client-go package (https://github.com/aerospike/aerospike-client-go).
+//
+// `WrapClient` will wrap an aerospike `Client` and return a new struct with all
+// the same methods, so should be seamless for existing applications. It also
+// has an additional `WithContext` method which can be used to connect a span
+// to an existing trace.
+package aerospike // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/aerospike/aerospike-client.v5"
+
+import (
+	"context"
+	"math"
+	"time"
+
+	as "github.com/aerospike/aerospike-client-go/v5"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+// WrapClient wraps an aerospike.Client so that all requests are traced using the
+// default tracer with the service name "aerospike".
+func WrapClient(client *as.Client, opts ...ClientOption) *Client {
+	cfg := new(clientConfig)
+	defaults(cfg)
+	for _, opt := range opts {
+		opt(cfg)
+	}
+	log.Debug("contrib/aerospike/aerospike-client.v5: Wrapping Client: %#v", cfg)
+	return &Client{
+		Client:  client,
+		cfg:     cfg,
+		context: context.Background(),
+	}
+}
+
+// A Client is used to trace requests to the aerospike server.
+type Client struct {
+	*as.Client
```

Comment:
For some reason I feel we should document this too, just in case someone gets ideas or gets confused.
```suggestion
	// Client specifies the original client which was wrapped. Do not use this client directly.
	// It will result in untraced commands.
	*as.Client
```